integration-tests/common-test-code/src/main/java/com/vaadin/flow/quarkus/it/uievents/NavigationObserver.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2000-2018 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.vaadin.flow.quarkus.it.uievents;
+
+import javax.enterprise.event.Observes;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.EventObject;
+import java.util.List;
+
+import com.vaadin.flow.router.AfterNavigationEvent;
+import com.vaadin.flow.router.BeforeEnterEvent;
+import com.vaadin.flow.router.BeforeLeaveEvent;
+import com.vaadin.quarkus.annotation.VaadinSessionScoped;
+
+@VaadinSessionScoped
+public class NavigationObserver implements Serializable {
+
+    private List<EventObject> navigationEvents = new ArrayList<>();
+
+    private void onBeforeLeave(@Observes BeforeLeaveEvent event) {
```

Comment:
I spent quite much time to keep `private` here.
I don't care about quarkus recommendation: this method may not be exposed since it breaks encapsulation.
This is a production code.